src/hooks/use-timeout.ts

```diff
@@ -0,0 +1,22 @@
+// h/t Dan Abramov - https://overreacted.io/making-setinterval-declarative-with-react-hooks/
+import { useEffect, useRef } from 'react'
+
+export function useInterval(callback: () => void, delay?: number | null): void {
+	const savedCallback = useRef<() => void>()
+
+	// Remember the latest callback.
+	useEffect(() => {
+		savedCallback.current = callback
+	}, [callback])
+
+	// Set up the interval.
+	useEffect(() => {
+		function tick() {
+			savedCallback.current?.()
+		}
+		if (delay !== null) {
```

Comment:
Does `delay` need to be able to be both `null` and `undefined`? `undefined` would pass this `if` I believe, not sure if that's intentional or not.

Reply:
Good catch. This code from Dan Abramov's [blog](https://overreacted.io/making-setinterval-declarative-with-react-hooks/), so you've officially found a bug in his code.

Reply:
🤗